symengine/pow.cpp

```diff
@@ -94,7 +91,14 @@ RCP<const Basic> pow(const RCP<const Basic> &a, const RCP<const Basic> &b)
         return pownum(rcp_static_cast<const Number>(b), zero);
     }
     if (eq(*b, *one)) return a;
-    if (eq(*a, *zero)) return zero;
+    if (eq(*a, *zero)) {
+    if (is_a_Number(*b)) {
+        if(rcp_static_cast<const Number>(b)->is_negative())
+            throw std::runtime_error("pow: Zero cannot be raised to negative power");
+        }
```

Comment:
Also add a check for `static_cast<const Number &>(*b).is_zero()` and throw an error that says indeterminate.